libs/utils/src/http/request.rs

```diff
@@ -29,6 +30,52 @@ pub fn parse_request_param<T: FromStr>(
     }
 }
 
+fn get_query_param<'a>(
+    request: &'a Request<Body>,
+    param_name: &str,
+) -> Result<Option<Cow<'a, str>>, ApiError> {
+    let query = match request.uri().query() {
+        Some(q) => q,
+        None => return Ok(None),
+    };
+    let mut values = url::form_urlencoded::parse(query.as_bytes()).filter_map(|(k, v)| {
+        if k == param_name {
+            Some(v)
+        } else {
+            None
+        }
+    });
```

Comment:
```suggestion
    }).fuse();
```

Since the `value1` might be None, it the second `.next` could do whatever.

Reply:
Or alternatively below:

```rust
let Some(value1) = values.next() else { return Ok(None) };
```

Reply:
> Since the value1 might be None, it the second .next could do whatever.

Good catch, thx!